engine/src/animation.rs

```diff
@@ -0,0 +1,334 @@
+use crate::assets::animation::*;
+use crate::prelude::*;
+
+pub struct AnimationPlugin;
+
+impl Plugin for AnimationPlugin {
+    fn build(&self, app: &mut App) {
+        app.add_system(sprite_animation_update.in_set(GameTickSet::Pre));
+    }
+}
+
+#[derive(Component)]
+pub struct SpriteAnimationPlayer {
+    animation: Handle<SpriteAnimation>,
+    state: SpriteAnimationPlayerState,
+}
+
+enum SpriteAnimationPlayerState {
+    Stopped,
+    PendingPlay {
```

Comment:
This can probably be simplified. I wrote it this way originally, to support a use case where the animation asset is unavailable (not preloaded) and needs to be loaded via the AssetServer on demand. However, on second thought, that might be undesirable. It might be better to just assume (and enforce) that animations are preloaded and always available.